src/Bicep.LangServer/Configuration/BicepConfigChangeHandlingConfigurationManager.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO.Abstractions;
+using Bicep.Core.Analyzers.Linter;
+using Bicep.Core.Configuration;
+using Bicep.LanguageServer.Telemetry;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+
+namespace Bicep.LanguageServer.Configuration
+{
+    public class BicepConfigChangeHandlingConfigurationManager : ConfigurationManager, IBicepConfigChangeHandler
+    {
+        private readonly ILinterRulesProvider linterRulesProvider;
+        private readonly ITelemetryProvider telemetryProvider;
+
+        public BicepConfigChangeHandlingConfigurationManager(
+            ILinterRulesProvider linterRulesProvider,
+            ITelemetryProvider telemetryProvider,
+            IFileSystem fileSystem) : base(fileSystem)
+        {
+            this.linterRulesProvider = linterRulesProvider;
+            this.telemetryProvider = telemetryProvider;
+        }
+
+        public void HandleBicepConfigOpenEvent(DocumentUri documentUri)
+        {
+            HandleBicepConfigOpenOrChangeEvent(documentUri);
+        }
+
+        public void HandleBicepConfigChangeEvent(DocumentUri documentUri)
+        {
+            HandleBicepConfigOpenOrChangeEvent(documentUri);
+            // The change may have rendered a config file invalid, or the event itself may have represented a file creation or deletion.
+            // In either case, the lookup cache would be stale.
+            PurgeLookupCache();
+        }
+
+        private void HandleBicepConfigOpenOrChangeEvent(DocumentUri documentUri) => RefreshConfigCacheEntry(documentUri.ToUri());
+
+        public void HandleBicepConfigSaveEvent(DocumentUri documentUri)
+        {
+            if (RefreshConfigCacheEntry(documentUri.ToUri()) is {} update)
+            {
+                TelemetryHelper.SendTelemetryOnBicepConfigChange(update.prevConfiguration, update.newConfiguration, linterRulesProvider, telemetryProvider);
+            }
+            // It's possible the saved document is invalid, which could render the lookup cache invalid as well.
+            PurgeLookupCache();
+        }
+
+        public void HandleBicepConfigCloseEvent(DocumentUri documentUri) 
+            => RemoveConfigCacheEntry(documentUri.ToUri());
+
+        protected override RootConfiguration GetDefaultConfiguration(bool loadErrorEncountered) => loadErrorEncountered
+            ? base.GetDefaultConfiguration(loadErrorEncountered).WithAllAnalyzersDisabled()
+            : base.GetDefaultConfiguration(loadErrorEncountered);
```

Comment:
1. Why do we need to override for different behavior in the language server?
1. Why disable analyzers if we hit a load error?

Reply:
This was the existing behavior in the language server, and my intention with this PR is to avoid any behavior changes outside of fixing the config loading bug.

Reply:
I removed this override in a separate commit (https://github.com/Azure/bicep/pull/8301/commits/c12487cd898310e4b00252c70268d056843f51c5) so that we can back it out if necessary. Whether we keep running analyzers when a model has unreadable config file is pretty edgy as far as edge cases go, so I'd be comfortable including it.